tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -768,3 +793,198 @@ def test_copy_obj_after_server_pod_restart(self):
                                                           f"{bkt}.")
         LOGGER.info("Step 8: Downloaded copied objects & verify etags.")
         LOGGER.info("COMPLETED: Test to verify copy object after server pod restart.")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-44851")
+    def test_obj_ver_during_server_pod_restart(self):
+        """
+        This test tests object versioning during server pod restart
+        """
+        LOGGER.info("STARTED: Test to verify object versioning during server pod restart.")
+        event = threading.Event()
+        get_output = Queue()
+        put_output = Queue()
+        LOGGER.info("Create bucket with %s name.", self.bucket_name)
+        resp = self.s3_test_obj.create_bucket(self.bucket_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Created bucket with %s name.", self.bucket_name)
+        LOGGER.info("Step 1: Upload object of %s size before enabling versioning.", self.f_size)
+        if os.path.exists(self.multipart_obj_path):
+            os.remove(self.multipart_obj_path)
+        system_utils.create_file(self.multipart_obj_path, b_size=self.f_size, count=1)
+        self.extra_files.append(self.multipart_obj_path)
+        args = {'chk_null_version': True, 'is_unversioned': True, 'file_path':
+            self.multipart_obj_path}
+        resp = self.ha_obj.parallel_put_object(event, self.s3_test_obj, self.bucket_name,
+                                               self.object_name, **args)
+        assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+        self.version_etag.update({self.bucket_name: []})
+        self.version_etag[self.bucket_name].extend(resp[1])
+        LOGGER.info("Step 1: Uploaded object of %s size before enabling versioning.", self.f_size)
+        LOGGER.info("Step 2: Enable versioning on %s.", self.bucket_name)
+        resp = self.s3_ver.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 2: Enabled versioning on %s.", self.bucket_name)
+        LOGGER.info("Step 3: Upload same object %s after enabling versioning, list & verify "
+                    "the same.", self.object_name)
+        args = {'file_path': self.multipart_obj_path}
+        resp = self.ha_obj.parallel_put_object(event, self.s3_test_obj, self.bucket_name,
+                                               self.object_name, **args)
+        assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+        self.version_etag[self.bucket_name].extend(resp[1])
+        resp = self.ha_obj.list_verify_version(self.s3_ver, self.bucket_name, self.version_etag[
+            self.bucket_name])
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 3: Uploaded same object %s after enabling versioning, listed & verified "
+                    "the same.", self.object_name)
```

Comment:
Is this during test or before and after test?

Reply:
During